src/components/CreateComment.jsx

```diff
@@ -0,0 +1,104 @@
+import { useContext, useState } from "react";
+import { useParams } from "react-router-dom";
+import TextField from "@mui/material/TextField";
+import CommentIcon from "@mui/icons-material/Comment";
+import Button from "@mui/material/Button";
+import Stack from "@mui/material/Stack";
+import { postComment } from "../api";
+import { UserContext } from "../contexts/User";
+import { Typography } from "@mui/material";
+
+export const CreateComment = ({ setArticleComments }) => {
+  const [showCommentInput, setShowCommentInput] = useState(false);
+  const [commentText, setCommentText] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const { article_id: article_id_from_params } = useParams();
+  const { user } = useContext(UserContext);
+
+  const handleTextChange = (event) => {
+    setCommentText(event.target.value);
+  };
+
+  const handleSubmitComment = () => {
+    if (article_id_from_params && user.username && commentText) {
+      // optimistc rendering stuff
+      // put comment into state, so that UI updates automatically
+      setArticleComments((prevComments) => {
+        return [
+          {
+            comment_id: 99999,
+            body: commentText,
+            article_id: article_id_from_params,
+            author: user.username,
+            votes: 0,
+            createdAt: new Date(),
+          },
+          ...prevComments,
+        ];
+      });
+
+      // send new comment to api, to update DB
+      // if user refreshes page, all comments will be loaded from api
+      setIsError(false);
+      setIsLoading(true);
+      postComment(article_id_from_params, user.username, commentText)
+        .then((data) => {
+          setShowCommentInput(!showCommentInput);
+          setIsLoading(false);
+          setCommentText("");
+          console.log(data);
+        })
+        .catch((error) => {
+          setIsLoading(false);
+          setIsError(error);
```

Comment:
This all looks good where you're setting the error, but at the moment the comment posts then when we refresh it isn't there. I think this is fine to have it optimistically rendering, but can we get some sort of message to the user like "Something went wrong" ?

Reply:
yep, have this on a list that i'm keeping for dealing with error handling when I've finished building everything. 🧘‍♀️

Reply:
also MUI makes styling WAAAAYYYYY easier than trying to wrap my brain around css 😂